root build fails in clad when using more than one job

### Check duplicate issues.

- [x] Checked for duplicates

### Description


When building root (have tried versions 6.33.01, 6.32.08, 6.32.02) with gcc13.3, building the clad component fails with an error of `bad file descriptor`, if `-jN` is used when `N > 1`:
```
> cat interpreter/cling/tools/plugins/clad/clad-prefix/src/clad-stamp/clad-build-err.log 
gmake[4]: *** read jobs pipe: Bad file descriptor.  Stop.
gmake[4]: *** Waiting for unfinished jobs....
gmake[5]: *** read jobs pipe: Bad file descriptor.  Stop.
gmake[5]: *** Waiting for unfinished jobs....
gmake[4]: *** [CMakeFiles/Makefile2:266: lib/Differentiator/CMakeFiles/cladDifferentiator.dir/all] Error 2
gmake[3]: *** [Makefile:136: all] Error 2
```
If I continue the build with just 1 job, it works fine. I can in fact kill the build after clad finishes building, and resume with -j20, and it will finish properly.

### Reproducer

```
cmake -Dxrootd=OFF -DCMAKE_INSTALL_PREFIX=/opt/root/v6-33-01_gcc133 ../src
```

### ROOT version

multiple. tried
6.33.01
6.32.08
6.32.02

### Installation method

git clone

### Operating system

linux alma 9

### Additional context

_No response_

--- a/interpreter/cling/tools/plugins/clad/CMakeLists.txt
+++ b/interpreter/cling/tools/plugins/clad/CMakeLists.txt
@@ -85,8 +85,12 @@ ExternalProject_Add(
              -DLLVM_DIR=${LLVM_BINARY_DIR}
              -DCLANG_INCLUDE_DIRS=${CLANG_INCLUDE_DIRS}
              ${_clad_extra_cmake_args}
-  BUILD_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS}
-  INSTALL_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS} --target install
+  # FIXME
+  # Building with 1 core is a temporary workaround for #16654 and has to be 
+  # there until the behaviour of the clad build on ubuntu 24.10 is understood.
+  # The performance penalty in the build is negligible.
+  BUILD_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS} -j 1
+  INSTALL_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS} -j 1 --target install
   BUILD_BYPRODUCTS ${CLAD_BYPRODUCTS}
   ${_clad_cmake_logging_settings}
   # We need the target clangBasic to be built before building clad. However, we

--- a/interpreter/cling/tools/plugins/clad/CMakeLists.txt
+++ b/interpreter/cling/tools/plugins/clad/CMakeLists.txt
@@ -24,7 +24,7 @@ if(NOT _clad_build_type STREQUAL "" AND NOT _clad_build_type STREQUAL ".")
   set(EXTRA_BUILD_ARGS --config ${_clad_build_type})
 endif()
 
-if(NOT MSVC AND CMAKE_VERSION VERSION_LESS 3.11.1)
+if(NOT MSVC AND CMAKE_VERSION VERSION_LESS 3.31.1)
   # Workaround for https://gitlab.kitware.com/cmake/cmake/-/issues/26398
   # The problem got eventually fixed upstream in CMake.
   list(APPEND EXTRA_BUILD_ARGS "-j 1")

--- a/interpreter/cling/tools/plugins/clad/CMakeLists.txt
+++ b/interpreter/cling/tools/plugins/clad/CMakeLists.txt
@@ -23,6 +23,13 @@ endif(MSVC)
 if(NOT _clad_build_type STREQUAL "" AND NOT _clad_build_type STREQUAL ".")
   set(EXTRA_BUILD_ARGS --config ${_clad_build_type})
 endif()
+
+if(NOT MSVC AND CMAKE_VERSION VERSION_LESS 3.11.1)
+  # Workaround for https://gitlab.kitware.com/cmake/cmake/-/issues/26398
+  # The problem got eventually fixed upstream in CMake.
+  list(APPEND EXTRA_BUILD_ARGS "-j 1")
+endif()
+
 set(_CLAD_LIBRARY_PATH ${CMAKE_CURRENT_BINARY_DIR}/clad-prefix/src/clad-build/${_clad_build_type}/lib${LLVM_LIBDIR_SUFFIX})
 
 # build byproducts only needed by Ninja
@@ -85,12 +92,8 @@ ExternalProject_Add(
              -DLLVM_DIR=${LLVM_BINARY_DIR}
              -DCLANG_INCLUDE_DIRS=${CLANG_INCLUDE_DIRS}
              ${_clad_extra_cmake_args}
-  # FIXME
-  # Building with 1 core is a temporary workaround for #16654 and has to be 
-  # there until the behaviour of the clad build on ubuntu 24.10 is understood.
-  # The performance penalty in the build is negligible.
-  BUILD_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS} -j 1
-  INSTALL_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS} -j 1 --target install
+  BUILD_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS}
+  INSTALL_COMMAND ${CMAKE_COMMAND} --build . ${EXTRA_BUILD_ARGS} --target install
   BUILD_BYPRODUCTS ${CLAD_BYPRODUCTS}
   ${_clad_cmake_logging_settings}
   # We need the target clangBasic to be built before building clad. However, we

